### PR TITLE
Add Kustomize to ko and ko-gcloud images

### PR DESCRIPTION
**Pull request by mnuttall**
# Changes

Add Kustomize to ko and ko-gcloud images. Needed until `ko -k` becomes a thing. 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [n/a] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by mnuttall** [comment]
We need to say `kustomize build [path] | ko resolve -f - > release.this.file.yaml` in the experimental and dashboard /tekton release automation code. We'd like to keep using our ko images and so hope that this small change will be acceptable. See #plumbing for a short discussion. Thank you!

**Comment by mnuttall** [comment]
I'm happy to use the same mechanism! https://github.com/kubernetes-sigs/kustomize/blob/master/docs/INSTALL.md offers a go-based install, a style we all ready use in ko-gcloud. If however we prefer the same `curl` in both then yes I can do that, of course.

**Comment by afrittoli** [comment]
goomba at your service! </p>Cloud not build the requested image. Please check the <a href="http://35.222.249.224/?buildid=39e1be27-5781-4416-aa87-a637e4b36359&namespace=mario">build logs</a>

**Comment by afrittoli** [comment]
tekton at your service! </p>Here is the image you requested: <a href="https://gcr.io/tekton-releases/dogfooding/ko:gcloud-mario">built image</a>|<a href="http://35.222.249.224/?buildid=5604f98d-6af4-442b-8faa-1accf1687822&namespace=mario">build logs</a>

**Comment by afrittoli** [comment]
> goomba at your service!
> Cloud not build the requested image. Please check the [build logs](http://35.222.249.224/?buildid=39e1be27-5781-4416-aa87-a637e4b36359&namespace=mario)

It looks like the golang image does not include `gcc` and the gcloud one does.

File: tekton/images/ko/Dockerfile [hunk 1]
```diff
@@ -16,4 +16,4 @@ LABEL maintainer "Tekton Authors <tekton-dev@googlegroups.com>"
 
 ENV GO111MODULE on
 RUN go get github.com/google/ko/cmd/ko@master
-
+RUN go install sigs.k8s.io/kustomize/kustomize/v3
```
Comment (afrittoli) [inline tekton/images/ko/Dockerfile]: This doesn't work here because the golang image does not include `gcc`

**Comment by afrittoli** [comment]
mario at your service! </p>Here is the image you requested: <a href="https://gcr.io/tekton-releases/dogfooding/ko:mario">built image</a>|<a href="http://35.222.249.224/?buildid=22cb267d-aa13-41a4-a226-d3b3edba5610&namespace=mario">build logs</a>

**Comment by afrittoli** [comment]
goomba at your service! </p>Cloud not build the requested image. Please check the <a href="http://35.222.249.224/?buildid=1eb6c816-1ae6-43cb-a576-39a3b6ae6807&namespace=mario">build logs</a>

**Comment by vdemeester** [comment]
> goomba at your service!
> Cloud not build the requested image. Please check the [build logs](http://35.222.249.224/?buildid=1eb6c816-1ae6-43cb-a576-39a3b6ae6807&namespace=mario)

oh boy… :sweat: 

```
[build-and-push]        sigs.k8s.io/kustomize/cmd/config@v0.0.0: reading sigs.k8s.io/kustomize/cmd/config/cmd/config/go.mod at revision cmd/config/v0.0.0: unknown revision cmd/config/v0.0.0
```

File: tekton/images/ko/Dockerfile (hunk 1 above)
Comment (mnuttall) [reply tekton/images/ko/Dockerfile]: Yes I had not realised just how minimal the `ko` image, based on Alpine, was. I've had to `apk add` `git`, `musl-dev` and `gcc` to get `kustomize` installed. I hope that's acceptable.

File: tekton/images/ko/Dockerfile [hunk 2]
```diff
@@ -17,3 +17,10 @@ LABEL maintainer "Tekton Authors <tekton-dev@googlegroups.com>"
 ENV GO111MODULE on
 RUN go get github.com/google/ko/cmd/ko@master
 
+RUN apk update
+RUN apk add --no-cache musl-dev
```
Comment (vdemeester) [inline tekton/images/ko/Dockerfile]: @mnuttall can we have those all on one line (`--no-cache` means we shouldn't need `apk update` before)

```
RUN apk add --no-cache musl-dev gcc git
```

:angel:

Comment (mnuttall) [reply tekton/images/ko/Dockerfile]: /mario build tekton/images/ko-gcloud ko:gcloud-mario

Comment (mnuttall) [reply tekton/images/ko/Dockerfile]: I don't know if Mario listens to me : /

**Comment by afrittoli** [comment]
goomba at your service! </p>Cloud not build the requested image. Please check the <a href="http://35.222.249.224/?buildid=c1f13b4e-04c2-4622-8465-512d62616bc3&namespace=mario">build logs</a>

**Comment by afrittoli** [comment]
goomba at your service! </p>Cloud not build the requested image. Please check the <a href="http://35.222.249.224/?buildid=f38366f8-2837-487a-bc0c-1683ca3d7c45&namespace=mario">build logs</a>

**Comment by afrittoli** [comment]
luigi at your service! </p>Here is the image you requested: <a href="https://gcr.io/tekton-releases/dogfooding/ko:mario">built image</a>|<a href="http://35.222.249.224/?buildid=2e90a1c6-ad41-4f9b-9659-205bdbed9c5f&namespace=mario">build logs</a>

**Comment by vdemeester** [comment]
Looking at the cluster, `mario` doesn't handle 2 command at the same time for now :stuck_out_tongue: 
/cc @afrittoli

**Comment by afrittoli** [comment]
mario at your service! </p>Here is the image you requested: <a href="https://gcr.io/tekton-releases/dogfooding/ko:gcloud-mario">built image</a>|<a href="http://35.222.249.224/?buildid=1a34d59d-e4cf-460b-8a12-4cc26a521249&namespace=mario">build logs</a>

**Comment by afrittoli** [comment]
luigi at your service! </p>Here is the image you requested: <a href="https://gcr.io/tekton-releases/dogfooding/ko:gcloud-mario">built image</a>|<a href="http://35.222.249.224/?buildid=dd3046de-1a8b-44dc-bd3b-fbba0f54719a&namespace=mario">build logs</a>

**Pull request merged**